itest/opstackl2/op_e2e_test.go

```diff
@@ -185,3 +185,58 @@ func TestFinalityStuckAndRecover(t *testing.T) {
 		"OP chain fianlity is recovered, the latest finalized block height %d",
 	), nextFinalizedHeight)
 }
+
+func TestFinalityGadgetServer(t *testing.T) {
+	// start the consumer manager
+	ctm := StartOpL2ConsumerManager(t, 2)
+	defer ctm.Stop(t)
+
+	// register, get BTC delegations, and start FPs
+	n := 2
+	fpList := ctm.SetupFinalityProviders(t, n, []stakingParam{
+		// for the first FP, we give it more power b/c it will be used later
+		{e2eutils.StakingTime, 3 * e2eutils.StakingAmount},
+		{e2eutils.StakingTime, e2eutils.StakingAmount},
+	})
+
+	// check both FPs have committed their first public randomness
+	// TODO: we might use go routine to do this in parallel
+	for i := 0; i < n; i++ {
+		e2eutils.WaitForFpPubRandCommitted(t, fpList[i])
+	}
+
+	// wait until the BTC staking is activated
+	l2BlockAfterActivation := ctm.waitForBTCStakingActivation(t)
+
+	// both FP will sign the first block
+	targetBlockHeight := ctm.WaitForTargetBlockPubRand(t, fpList, l2BlockAfterActivation)
+	ctm.WaitForFpVoteReachHeight(t, fpList[0], targetBlockHeight)
+	ctm.WaitForFpVoteReachHeight(t, fpList[1], targetBlockHeight)
+	t.Logf(log.Prefix("Both FP instances signed the first block"))
+
+	// both FP will sign the second block
+	ctm.WaitForFpVoteReachHeight(t, fpList[0], targetBlockHeight+1)
+	ctm.WaitForFpVoteReachHeight(t, fpList[1], targetBlockHeight+1)
+	t.Logf(log.Prefix("Both FP instances signed the second block"))
+
+	// start finality gadget processing blocks
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		t.Logf(log.Prefix("Starting finality gadget"))
+		err := ctm.FinalityGadget.ProcessBlocks(ctx)
+		require.NoError(t, err)
+	}()
+
+	// check latest block
+	require.Eventually(t, func() bool {
+		block, err := ctm.FinalityGadget.QueryLatestFinalizedBlock()
+		if block == nil {
+			return false
+		}
+		require.NoError(t, err)
+		return block.BlockHeight > targetBlockHeight+6
```

Comment:
does it have to be 6 or it can be a random number? (if it's random, let's add a comment above to document that)

Reply:
will add a comment to explain